packages/browser/src/plugins/ajs-destination/index.ts

```diff
@@ -116,11 +117,11 @@ export class LegacyDestination implements DestinationPlugin {
   }
 
   ready(): Promise<unknown> {
-    return this.onReady ?? Promise.resolve()
+    return this.readyPromise.promise
   }
 
   async load(ctx: Context, analyticsInstance: Analytics): Promise<void> {
-    if (this._ready || this.onReady !== undefined) {
+    if (this._ready || this.readyPromise.settled) {
```

Comment:
You could bypass the need to add `settled` by having `_ready` be `boolean | undefined`. Set it to `true` or `false` at `resolve` or `reject`. If it is `undefined` it is still unsettled.

Reply:
Or better yet could be modeled into an stateflow-like enum: Created -> Initialized -> Ready (+ error states)

Reply:
Please see the latest push for revision to this